modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements/evidence_submissions_controller.rb

```diff
@@ -30,20 +41,29 @@ def show
 
         private
 
-        def set_notice_of_disagreement
-          return unless params[:nod_id]
+        def nod_id_present?
+          nod_id_missing_error unless params[:nod_id]
+        end
+
+        def nod_id_missing_error
+          error = { title: 'bad_request', detail: I18n.t('appeals_api.errors.missing_nod_id') }
+          log_error(error)
 
-          @notice_of_disagreement ||= AppealsApi::NoticeOfDisagreement.find_by(id: params[:nod_id])
-          raise Common::Exceptions::RecordNotFound, params[:nod_id] unless @notice_of_disagreement
+          render json: { errors: [error] }, status: :bad_request
```

Comment:
It's been a while since I've dealt with this, but you might want to check the interaction with the double render from the private methods. Does this render short circuit the controller execution?

Reply:
IIRC because it's in a `before_action` it will recognize the render there and short circuit before it gets to the #create aciton.